packages/langium/src/lsp/completion/completion-provider.ts

```diff
@@ -400,9 +400,10 @@ export class DefaultCompletionProvider implements CompletionProvider {
                 const scope = this.scopeProvider.getScope(refInfo);
                 const duplicateStore = new Set<string>();
                 scope.getAllElements().forEach(e => {
-                    if (!duplicateStore.has(e.name) && this.filterCrossReference(e)) {
+                    const storeId = e.documentUri + e.path;
```

Comment:
`${e.documentUri}#${e.path}`

Reply:
Fragment path starts with an `/` which is already a good separator as doc URI normally ends with a file extension I thought. That why I left out an additional separator, but yes we can add `#`.